worker/changestream/eventmultiplexer/notifyworker.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package eventmultiplexer
+
+import (
+	"sync/atomic"
+
+	"github.com/juju/juju/core/changestream"
+	"gopkg.in/tomb.v2"
+)
+
+type Signaller interface {
+	Signal(changes []changestream.ChangeEvent)
+}
+
+type notifyWorker struct {
+	tomb tomb.Tomb
+
+	notary      chan struct{}
+	numNotaries int64
+}
+
+func newNotifyWorker(size int64) *notifyWorker {
+	w := &notifyWorker{
+		numNotaries: size,
+		notary:      make(chan struct{}),
+	}
+	w.tomb.Go(w.loop)
+	return w
+}
+
+func (s *notifyWorker) Notify(sub *subscription, changes []changestream.ChangeEvent) {
+	s.tomb.Go(func() error {
+		sub.Signal(changes)
+		atomic.AddInt64(&s.numNotaries, -1)
+
+		select {
+		case <-s.tomb.Dying():
+			return tomb.ErrDying
+		case s.notary <- struct{}{}:
```

Comment:
Probably not really a problem, but this go routine may hang around a little longer than expected because the loop isn't receiving from the notary channel anymore after numNotaries reaches 0. So in this case we rely on the tomb actually dying to cleanup.